pkg/util/ingress/ingress.go

```diff
@@ -259,31 +259,46 @@ func ValidateIngressPath(copyIng *networkingv1.Ingress, disablePathTypeValidatio
 	}
 
 	for _, rule := range copyIng.Spec.Rules {
+
 		if rule.HTTP == nil {
 			continue
 		}
-		if err := checkPath(rule.HTTP.Paths, disablePathTypeValidation, regexPath); err != nil {
+
+		if err := checkPath(rule.HTTP.Paths, enablePathTypeValidation, regexPath); err != nil {
 			return fmt.Errorf("error validating ingressPath: %w", err)
 		}
 	}
 	return nil
 }
 
-func checkPath(paths []networkingv1.HTTPIngressPath, disablePathTypeValidation bool, regexSpecificChars *regexp.Regexp) error {
+func checkPath(paths []networkingv1.HTTPIngressPath, enablePathTypeValidation bool, regexSpecificChars *regexp.Regexp) error {
+
 	for _, path := range paths {
 		if path.PathType == nil {
 			path.PathType = &defaultPathType
 		}
 
-		if disablePathTypeValidation || *path.PathType == networkingv1.PathTypeImplementationSpecific {
+		klog.V(9).InfoS("PathType Validation", "enablePathTypeValidation", enablePathTypeValidation, "regexSpecificChars", regexSpecificChars.String(), "Path", path.Path)
+
+		switch pathType := *path.PathType; pathType {
+		case networkingv1.PathTypeImplementationSpecific:
+
 			if !regexSpecificChars.MatchString(path.Path) {
 				return fmt.Errorf("path %s of type %s contains invalid characters", path.Path, *path.PathType)
 			}
-			continue
-		}
 
-		if !pathAlphaNumericRegex(path.Path) {
-			return fmt.Errorf("path %s of type %s contains invalid characters", path.Path, *path.PathType)
+		case networkingv1.PathTypeExact, networkingv1.PathTypePrefix:
+			if enablePathTypeValidation {
+				if !pathAlphaNumericRegex(path.Path) {
+					return fmt.Errorf("path %s of type %s contains invalid characters", path.Path, *path.PathType)
+				}
+				continue
+			}
+			if !regexSpecificChars.MatchString(path.Path) {
+				return fmt.Errorf("path %s of type %s contains invalid characters", path.Path, *path.PathType)
+			}
```

Comment:
This confuses me, if we don't enable pathType validation, we also check special chars?

Reply:
yes, the default list or the list provided by `PathAdditionalAllowedChars` 

I updated the comments and made the vars a little easier to follow the logic. 

Implementation specific, always check the regex 
Exactm Prefix, if validation, check that is only alpha numeric, otherwise check the regex list.